pymemcache/codecs.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2012 Pinterest.com
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import logging
+from io import BytesIO
+import six
+from six.moves import cPickle as pickle
+
+log = logging.getLogger(__name__)
+
+try:
+    long_type = long  # noqa
+except NameError:
+    long_type = None
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ICodec(object):
+    """
+    Interface for serializers.
+    """
+
+    @abc.abstractmethod
+    def from_python(self, key, value):
+        """
+        Serialize a python object.
+
+        :param str|unicode key: Key
+        :param str|unicode value: Value
+        :return tuple[str, int]: tuple(value, flags)
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def to_python(self, key, value, flags):
+        """
+        Deserialize a value into a python object.
+
+        :param str|unicode key: Key
+        :param str|unicode value: Value
+        :param int flags: Bitflag containing flags used to specify how to
+        deserialize this object.
+        :return object: Deserialized python object.
+        """
+        raise NotImplementedError()
+
+
+class Serde(ICodec):
+    """
+    Serialization handler.
+
+    Meant to be compatible with `python-memcached`.
+    """
+
+    FLAG_BYTES = 0
+    FLAG_PICKLE = 1 << 0
+    FLAG_INTEGER = 1 << 1
+    FLAG_LONG = 1 << 2
+    FLAG_COMPRESSED = 1 << 3  # unused, to main compatibility with python-memcached
+    FLAG_TEXT = 1 << 4
+
+    pickle_version = 0
+
+    def __init__(self, pickle_version=0):
+        """
+        Init
+
+        :param int pickle_version: Pickle version to use (from python).
+          Use `-1` to use the highest supported at runtime.
+          Deserialization is not affected by this parameter.
+
+          A forewarning with `0` (the default): If somewhere in your value lies
+          a slotted object, ie defines `__slots__`, even if you do not include
+          it in your pickleable state via `__getstate__`, python will raise:
+            ```
+            TypeError: a class that defines __slots__ without defining
+            __getstate__ cannot be pickled
+            ```
+        """
+        if pickle_version is not None:
+            self.pickle_version = pickle_version
+
+    def from_python(self, key, value):
```

Comment:
I changed it because in my experience serialize and deserialize are not the best names. Using a common anchor it seems to stick with more juniors easier and avoids headache. Still, I can update if you want, but I'd recommend not.